[cpp,hl/c] native compilation error for Void arguments

The following was failing with a `VerifyError` on the Flash target and now is failing on native compilation on C++:

```haxe
function f(arg:Void) {}
```

This is of course not normally written, but it can happen that `Void` is inferred through type parameters when void-returning callbacks are involved, like in this simplified example:

```haxe
class C<T> {
	var v:T;
	public function new(v:T) this.v = v;
	public function next<S>(f:()->S):C<S> return new C(f());
	public function handle(cb:T->Void) {cb(v);}
}

function returnVoid() {}

function main() {
	new C().next(returnVoid).handle(_ -> {});
}
```

Since this is passing haxe type checking all the way to the native compilation, it would be nice to get it fixed.

We can consider this a preparation for https://github.com/HaxeFoundation/haxe-evolution/pull/76. For Flash for now I just made the generator generate the wildcard type (`*`) when Void is used for non-return-type. Maybe something similar could/should be done for C++.

Also an issue on C++:
```
include/unit/issues/Issue9678.h:49:36: error: argument may not have 'void' type
                static void explicitVoidArg(void arg);
```

--- a/src/generators/genswf9.ml
+++ b/src/generators/genswf9.ml
@@ -247,7 +247,7 @@ let rec type_id ctx t =
 
 let type_opt ctx t =
 	match follow_basic t with
-	| TDynamic _ | TMono _ -> None
+	| TDynamic _ | TMono _ | TAbstract ({a_path = [],"Void"},_) -> None
 	| _ -> Some (type_id ctx t)
 
 let type_void ctx t =

--- a/tests/unit/src/unit/issues/Issue9678.hx
+++ b/tests/unit/src/unit/issues/Issue9678.hx
@@ -1,6 +1,7 @@
 package unit.issues;
 
 class Issue9678 extends unit.Test {
+	#if !cpp
 	function test() {
 		var called = 0;
 		function returnVoid() {
@@ -11,6 +12,7 @@ class Issue9678 extends unit.Test {
 	}
 
 	@:keep static function explicitVoidArg(arg:Void) {}
+	#end
 }
 
 private class C<T> {

--- a/tests/unit/src/unit/issues/Issue9678.hx
+++ b/tests/unit/src/unit/issues/Issue9678.hx
@@ -1,0 +1,21 @@
+package unit.issues;
+
+class Issue9678 extends unit.Test {
+	function test() {
+		var called = 0;
+		function returnVoid() {
+			called++;
+		}
+		new C(42).next(returnVoid).handle(_ -> called++);
+		eq(2, called);
+	}
+
+	@:keep static function explicitVoidArg(arg:Void) {}
+}
+
+private class C<T> {
+	final v:T;
+	public function new(v:T) this.v = v;
+	public function next<S>(f:()->S):C<S> return new C(f());
+	public function handle(cb:T->Void) {cb(v);}
+}

